readthedocs/profiles/views.py

```diff
@@ -78,8 +78,9 @@ def get_object(self):
         return self.request.user
 
     def form_valid(self, form):
-        self.request.user.delete()
+        user = self.get_object()
         logout(self.request)
+        user.delete()
```

Comment:
Why was this change needed?

Reply:
this was needed, it was given 500 if the user was deleted before the logout